client/instrumenter/src/main/java/org/hawkular/apm/instrumenter/ClientManager.java

```diff
@@ -76,24 +86,40 @@ public static void initialize(Retransformer trans) {
         }
 
         if (configService != null) {
-            // Read configuration
-            CollectorConfiguration config = configService.getCollector(null, null, null, null);
-
-            if (config != null) {
-                try {
-                    updateInstrumentation(config);
-                } catch (Exception e) {
-                    System.err.println("Failed to update instrumentation rules: " + e);
-                    e.printStackTrace();
-                }
-            } else {
-                log.severe("Unable to configure APM");
-            }
+            initConfig();
         } else {
             log.info("APM has not been configured");
         }
     }
 
+    protected static void initConfig() {
+        // Read configuration
+        CollectorConfiguration config = configService.getCollector(null, null, null, null);
+
+        if (config != null) {
+            try {
+                updateInstrumentation(config);
+            } catch (Exception e) {
+                log.severe("Failed to update instrumentation rules: " + e);
```

Comment:
This code executes in the javaagent - so throwing an exception would prevent the application from starting up. Not sure if that is desirable?
Currently only log severe message - but have a separate jira regarding logging admin issues to the server, so possibly this error could be sent to the server at that point, where an operator could take action.